ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -262,6 +262,41 @@ let test_cbt =
       )
   in
 
+
+  let test_vdi_data_destroy () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+
+    (* change VDI properties so that data_destroy passes *)
+    let pass_data_destroy vdi =
+      let sr = Db.VDI.get_SR ~__context ~self:vdi in
+      Db.SR.set_is_tools_sr ~__context ~self:sr ~value:false;
+      Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:true;
+      Db.VDI.set_cbt_enabled ~__context ~self:vdi ~value:true in
+
+    (* change VDI properties one by one so data_destroy only fails for the indicated reason *)
+    List.iter (fun (vdi_fun, api_error) ->
+        run_assert_equal_with_vdi ~__context ~vdi_fun `data_destroy (Some api_error))
+
+      [
+        (fun vdi -> pass_data_destroy vdi;
+          Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:false
+        ) ,         (Api_errors.operation_not_allowed, ["VDI is not a snapshot"]) ;
```

Comment:
Here just say
```ocaml
(Api_errors.operation_not_allowed, []) ;
```
as in the above tests - the function does not compare these string parameters of the error, to avoid breaking the test in case we change the strings we pass to the error. (Same for all the other ones below.)

Reply:
And we prefer to avoid doing too much formatting, and leave formatting to ocp-indent, so it's better to have on space after the `), ` in my opinion.